concent_api/core/validation.py

```diff
@@ -127,20 +127,21 @@ def validate_all_messages_identical(golem_messages_list: List[message.Message])
     base_golem_message = golem_messages_list[0]
 
     for i, golem_message in enumerate(golem_messages_list[1:], start=1):
-        for slot in base_golem_message.__slots__:
-            if getattr(base_golem_message, slot) != getattr(golem_message, slot):
-                raise ConcentValidationError(
-                    '{} messages are not identical. '
-                    'There is a difference between messages with index 0 on passed list and with index {}'
-                    'The difference is on field {}: {} is not equal {}'.format(
-                        type(base_golem_message).__name__,
-                        i,
-                        slot,
-                        getattr(base_golem_message, slot),
-                        getattr(golem_message, slot),
-                    ),
-                    error_code=ErrorCode.MESSAGES_NOT_IDENTICAL,
-                )
+        if not base_golem_message.__eq__(golem_message):
```

Comment:
Was the problem here related with nested messages in TTC?

Reply:
It was z problem with `encrypted` field in golem_messages. This field can be different and this method allows us to use method from GM and we will always know for sure is it equaled or not :)